src/redis-benchmark.c

```diff
@@ -632,6 +632,9 @@ static void writeHandler(aeEventLoop *el, int fd, void *privdata, int mask) {
                         fprintf(stderr, "Error writing to the server: %s\n", strerror(errno));
                     freeClient(c);
                     return;
+                } else if (nwritten > 0) {
+                    c->written += nwritten;
```

Comment:
i see this is the only place that sets it to anything other than zero (i.e. the code was really incomplete).
don't we need to also zero it on success?
can you track down the commit that introduced this mechanism and the bug?

Reply:
It's PR #7959 I think

Reply:
I saw a link that might be relevant: #9370

Reply:
@yoav-steinberg @filipecosta90 i see you where involved with #9370 but it was forgotten.
please take a look and comment what's the preferred way to put this behind us.

Reply:
This seems fine, and closer to what we need than the current state of #9370.

Reply:
@yoav-steinberg don't we need to reset it to 0 when we succeeded to write the last fragment?
or is it reset to 0 somewhere else?

Reply:
`resetClient()` zeros written. The benchmark creates/resets the client per write (or pipeline size write) operation.